lib/extractors/exported.js

```diff
@@ -1,40 +1,93 @@
 var traverse = require('babel-traverse').default,
-  isJSDocComment = require('../../lib/is_jsdoc_comment');
-
+  isJSDocComment = require('../../lib/is_jsdoc_comment'),
+  t = require('babel-types'),
+  nodePath = require('path'),
+  fs = require('fs');
 
 /**
  * Iterate through the abstract syntax tree, finding ES6-style exports,
  * and inserting blank comments into documentation.js's processing stream.
  * Through inference steps, these comments gain more information and are automatically
  * documented as well as we can.
+ * @param {Function} parseToAst funtiont that parses to an ast
  * @param {Object} ast the babel-parsed syntax tree
+ * @param {Object} data the name of the file
  * @param {Function} addComment a method that creates a new comment if necessary
  * @returns {Array<Object>} comments
  * @private
  */
-function walkExported(ast, addComment) {
+function walkExported(parseToAst, ast, data, addComment) {
   var newResults = [];
+  var filename = data.file;
+
+  function addBlankComment(data, path, node) {
+    return addComment(data, '', node.loc, path, node.loc, true);
+  }
+
+  function getComments(data, path) {
+    if (!hasJSDocComment(path)) {
+      return [addBlankComment(data, path, path.node)];
+    }
+    return path.node.leadingComments.filter(isJSDocComment).map(function (comment) {
+      return addComment(data, comment.value, comment.loc, path, path.node.loc, true);
+    }).filter(Boolean);
+  }
 
-  function addBlankComment(path, node) {
-    return addComment('', node.loc, path, node.loc, true);
+  function addComments(data, path, overrideName) {
+    var comments = getComments(data, path);
+    if (overrideName) {
+      comments.forEach(function (comment) {
+        comment.name = overrideName;
+      });
+    }
+    newResults.push.apply(newResults, comments);
   }
 
   traverse(ast, {
-    enter: function (path) {
-      if (path.isExportDeclaration()) {
-        if (!hasJSDocComment(path)) {
-          if (!path.node.declaration) {
-            return;
-          }
-          const node = path.node.declaration;
-          newResults.push(addBlankComment(path, node));
+    ExportDeclaration: function (path) {
```

Comment:
this could probably have an enter+skip to reduce the amount of traversal done here.